src/mantarray_desktop_app/process_monitor.py

```diff
@@ -307,6 +314,30 @@ def _check_and_handle_ok_comm_to_main_queue(self) -> None:
                 ch_index = int(parsed_description[2][-1])
                 offset_val = communication["wire_out_value"]
                 self._add_offset_to_shared_dict(adc_index, ch_index, offset_val)
+        elif communication_type == "barcode_comm":
+            if "barcodes" not in self._values_to_share_to_server:
+                self._values_to_share_to_server["barcodes"] = dict()
+            board_idx = communication["board_idx"]
+            if board_idx not in self._values_to_share_to_server["barcodes"]:
+                self._values_to_share_to_server["barcodes"][board_idx] = dict()
```

Comment:
in the future, you may be able to make use of a `defaultdict` for situations like this